backend/app/status_manager.py

```
@@ -0,0 +1,203 @@
+import logging
+from pydantic import BaseModel, Field
+from typing import Optional, Dict
```

Ruff (0.8.2)

`typing.Dict` is deprecated, use `dict` instead

(UP035)